pkg/operatorsource/handler.go

```diff
@@ -16,36 +25,31 @@ type Handler interface {
 }
 
 type handler struct {
-	reconciler Reconciler
+	strategy     PhaseReconcilerStrategy
+	kubeclient   Kubeclient
+	transitioner PhaseTransitioner
 }
 
 func (h *handler) Handle(ctx context.Context, event sdk.Event) error {
-	opsrc := event.Object.(*v1alpha1.OperatorSource)
-
-	if event.Deleted {
-		logrus.Infof("No action taken, object has been deleted [type=%s object=%s/%s]",
-			opsrc.TypeMeta.Kind, opsrc.ObjectMeta.Namespace, opsrc.ObjectMeta.Name)
-
-		return nil
+	in, ok := event.Object.(*v1alpha1.OperatorSource)
+	if !ok {
+		return fmt.Errorf("casting failed, wrong type provided")
 	}
 
-	reconciled, err := h.reconciler.IsAlreadyReconciled(opsrc)
+	reconciler, err := h.strategy.GetPhaseReconciler(event)
 	if err != nil {
 		return err
 	}
 
-	if reconciled {
-		logrus.Infof("Already reconciled, no action taken [type=%s object=%s/%s]",
-			opsrc.TypeMeta.Kind, opsrc.ObjectMeta.Namespace, opsrc.ObjectMeta.Name)
-
-		return nil
+	out, status, err := reconciler.Reconcile(ctx, in)
+	if err != nil {
+		return err
 	}
 
-	if err := h.reconciler.Reconcile(opsrc); err != nil {
-		return err
+	changed := h.transitioner.TransitionInto(out, status)
```

Comment:
Nit: Going with my comment above, you can avoid declaring the variable `changed` here and do the check here itself.